### PR TITLE
Make the ciphertext a public input of the circuit

### PR DESCRIPTION
**Pull request by jrchatruc**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -79,9 +81,17 @@ pub fn encrypt(
         })?);
     }
 
-    let ciphertext = encrypt_and_generate_constraints(
+    let mut ciphertext_circuit: Vec<UInt8Gadget> = Vec::with_capacity(ciphertext.len());
+    for byte in ciphertext {
+        ciphertext_circuit.push(UInt8Gadget::new_input(constraint_system.clone(), || {
+            Ok(byte)
+        })?);
+    }
```
Comment (ilitteri) [inline src/lib.rs:91]: We could do this here.
```suggestion
    let ciphertext_circuit = UInt8Gadget::new_input_vec(constraint_system, ciphertext)?;
```

File: src/lib.rs [hunk 2]
```diff
@@ -133,9 +168,18 @@ pub fn synthesize_keys(plaintex_length: usize) -> Result<(ProvingKey, VerifyingK
         })?);
     }
 
+    let mut ciphertext_circuit: Vec<UInt8Gadget> =
+        Vec::with_capacity(default_ciphertext_input.len());
+    for byte in default_ciphertext_input {
+        ciphertext_circuit.push(UInt8Gadget::new_input(constraint_system.clone(), || {
+            Ok(byte)
+        })?);
+    }
```
Comment (ilitteri) [inline src/lib.rs:181]: Same here.
```suggestion
    let ciphertext_circuit = UInt8Gadget::new_input_vec(constraint_system, default_ciphertext_input)?;
```

File: src/lib.rs [hunk 3]
```diff
@@ -97,26 +107,51 @@ pub fn encrypt(
...
     let universal_srs =
         simpleworks::marlin::generate_universal_srs(1_000_000, 250_000, 3_000_000, rng)?;
     let constraint_system = ConstraintSystem::<ConstraintF>::new_ref();
 
-    let default_message_input = vec![0_u8; plaintex_length];
+    let default_message_input = vec![0_u8; plaintext_length];
     let default_secret_key_input = [0_u8; 16];
+    let default_ciphertext_input = vec![0_u8; plaintext_length];
 
     let mut message_circuit: Vec<UInt8Gadget> = Vec::with_capacity(default_message_input.len());
```
Comment (ilitteri) [inline src/lib.rs:160]: Here we could use `new_witness_vec`.

Comment (ilitteri) [reply src/lib.rs:160]: The same for `secret_key_circuit`.

**Review by ilitteri — COMMENTED** [review]
Left some comments

**Comment by jrchatruc** [comment]
Addressed comments here [14f7ea6](https://github.com/lambdaclass/zk_encryption_circuit/pull/26/commits/14f7ea64e346e123515028492f12b274b71aa70a)

**Comment by jrchatruc** [comment]
> 

Welp, ended up reverting the commit and adding `TODO`s because for some reason using the `_vec` variants breaks the integration tests

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**